File: h2_collect_results.R
INPUT.PATHS <- snakemake@input
RESULTS.DAT.OUT.PATH <- snakemake@output[[1]]

results.list <- lapply(INPUT.PATHS, read.table, header=TRUE, sep="\t", nrows=1)
result.filenames <- sapply(INPUT.PATHS, basename)

for(i in seq_along(results.list)){
  stopifnot(identical(colnames(results.list[[1]]), colnames(results.list[[i]])))
  stopifnot(results.list[[i]]$category == "L2_0")
}

results.dat <- as.data.frame(do.call(rbind, results.list))
results.dat$result.filename <- result.filenames

write.table(results.dat, file=RESULTS.DAT.OUT.PATH, col.names=TRUE, row.names=FALSE, quote=FALSE, sep="\t")

File: h2_collect_results.py
import pandas as pd
import os 

INPUT_PATHS=snakemake.input
OUT_DAT_PATH=snakemake.output[0]

#for testing
#INPUT_PATHS= []
#h2_dir = "/hpcdata/sg/sg_data/users/rachmaninoffn2/scratch/h2_test/h2_out/"
#OUT_DAT_PATH= "/hpcdata/sg/sg_data/users/rachmaninoffn2/scratch/h2_test/h2_compiled_results/pandas_test.tsv"
#all_files = os.listdir(h2_dir)
#
#for file in all_files:
#    if file.endswith(".results"):
#        INPUT_PATHS.append(h2_dir + "/" + file)


#read in data
df_list = []
for path in INPUT_PATHS:
    df = pd.read_csv(path, sep = "\t")
    df = df.assign(results_filename = os.path.basename(path))
    df_list.append(df)


#append into single dataframe
n_df = len(df_list)

out_df = df_list[0]
for i in range(1, n_df):
    out_df=out_df.append(df_list[i])

#write dataframe out
out_df.to_csv(OUT_DAT_PATH, sep = "\t", index = False)

